resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -356,7 +356,18 @@ protected void initChannel(DatagramChannel ch) throws Exception {
         });
 
         channelFuture = responseHandler.channelActivePromise;
-        ch = (DatagramChannel) b.register().channel();
+        ChannelFuture future = b.register();
+        Throwable cause = future.cause();
+        if (cause != null) {
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            }
+            if (cause instanceof Error) {
+                throw (Error) cause;
+            }
+            throw new IllegalStateException("Unable to create / register Channel", cause);
```

Comment:
ISE is for calling methods out of order.  I believe this should be a plain RuntimeException.

Reply:
@carl-mastrangelo hmm not sure about it... But not strong either. If you think this would be more correct please provide a pr :)